lib/Backends/CPU/CPUFunction.cpp

```diff
@@ -64,8 +68,13 @@ void CPUFunction::copyOutputsFromDevice(Context &ctx) {
 }
 
 void CPUFunction::freeAllocations() {
-  alignedFree(baseMutableWeightVarsAddress_);
-  alignedFree(baseActivationsAddress_);
+  if (baseMutableWeightVarsAddress_) {
+    alignedFree(baseMutableWeightVarsAddress_);
```

Comment:
It's safe to call `free` with null... but maybe alignedFree is different.  Not that it matters too much :)

Reply:
I'd not risk, overhead seems pretty low here :)